advanced/postgres_lambda_strategy/lambda_src/sql.py

```diff
@@ -0,0 +1,65 @@
+import logging
+
+from psycopg2 import sql
+
+from config import Config
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+def format_sql(username: str, event: dict, config: Config) -> sql.Composable:
+    """
+    Get the right sql statement for the supplied event and user
+    """
+    target_role = resolve_role(event)
+    logger.debug("Target role: %s", target_role)
+    event_type = event["event"]["type"]
+    if event_type == "escalate":
+        return format_grant(target_role, username)
+    elif event_type == "deescalate":
+        return format_revoke(target_role, username)
+    else:
+        raise RuntimeError(f"Unsupported event type: {event_type}")
+
+
+def resolve_role(event: dict) -> str:
+    """Get the role name from the target name"""
+    full_name = event["fields"]["target"]["name"]
+    return full_name.split("-")[-1]
```

Comment:
nit: This method seems to expect a target name to be prefixed with an environment, but in this example, there is only one target `readonly`. Should we simplify this method to just use the target name?